excellent/functions/builtin.go

```diff
@@ -2101,6 +2103,28 @@ func ExtractObject(env envs.Environment, args ...types.XValue) types.XValue {
 	return types.NewXObject(result)
 }
 
+// Keys takes an object and returns its properties
+//
+//	@(keys(object("a", 123, "b", "hello", "c", "world"))) -> [a, b, c]
+//	@(keys("string")) -> ERROR
+//	@(keys(10)) -> ERROR
+//	@(keys(null)) -> ERROR
+//
+// @function keys(object)
+func Keys(env envs.Environment, object *types.XObject) types.XValue {
+
+	if object == types.XObjectEmpty {
```

Comment:
should be valid to check an empty object.. return value is just `[]`

Reply:
nil is transformed to `XObjectEmpty `

https://github.com/nyaruka/goflow/blob/38f42b8822dcd533045051a42d0ddaccc48f2f7e/excellent/types/object.go#L241

So that as well return []

Reply:
I forgot that.. well ok